mmv1/products/netapp/backupVault.yaml

```diff
@@ -0,0 +1,87 @@
+# Copyright 2023 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Resource
+name: 'backupVault'
+description: |
+  A backup vault is the location where backups are stored. You can only create one backup vault per region.
+  A vault can hold multiple backups for multiple volumes in that region.
+references: !ruby/object:Api::Resource::ReferenceLinks
+  guides:
+    'Documentation': 'https://cloud.google.com/netapp/volumes/docs/protect-data/about-volume-backups'
+  api: 'https://cloud.google.com/netapp/volumes/docs/reference/rest/v1/projects.locations.backupVaults'
+base_url: projects/{{project}}/locations/{{location}}/backupVaults
+self_link: projects/{{project}}/locations/{{location}}/backupVaults/{{name}}
+create_url: projects/{{project}}/locations/{{location}}/backupVaults?backupVaultId={{name}}
+create_verb: :POST
+update_url: projects/{{project}}/locations/{{location}}/backupVaults/{{name}}
+update_verb: :PATCH
+update_mask: true
+delete_url: projects/{{project}}/locations/{{location}}/backupVaults/{{name}}
+delete_verb: :DELETE
+autogen_async: true
+async: !ruby/object:Api::OpAsync
+  operation: !ruby/object:Api::OpAsync::Operation
+    base_url: '{{op_id}}'
+id_format: 'projects/{{project}}/locations/{{location}}/backupVaults/{{name}}'
+import_format: ['projects/{{project}}/locations/{{location}}/backupVaults/{{name}}']
+parameters:
+  - !ruby/object:Api::Type::String
+    name: 'location'
+    required: true
+    immutable: true
+    url_param_only: true
+    description: |
+      Location (region) of the backup vault.
+  - !ruby/object:Api::Type::String
+    name: 'name'
+    description: |
+      The resource name of the backup vault. Needs to be unique per location.
+    required: true
+    immutable: true
+    url_param_only: true
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: 'netapp_backup_vault'
+    primary_resource_id: 'test_backup_vault'
+    skip_test: true
```

Comment:
Any reasons we skip the auto gen test?

Reply:
Yes. There is only one backup vault per region allowed. If we don't skip that test, it creates an autp-generated test which runs in parallel to my custom test and two resources are created in the same region, which will fail. There are two ways around it:
1. Run example test and custom test in different regions
2. Skip custom test, which was the recommendation from @shuyama1 . But I am open to enable it and run one of the test in a different region (need recommendation which one to use).

Reply:
@okrause @ScottSuarez yeah, I recommended to skip the generated test if it's covered by the custom update test, which seems the case for the KMS PR when I was checking, and it can potentially avoid running redundant tests as well as the regional resource limitation. If it's not the case here and we need to run both the generated and the update tests, we'll probably need to use different regions (any region should work, I assume) 
Sorry for the confusion and hope this will clarify things :)

Reply:
makes sense